### ISSUE TITLE
[Create/edit survey] Allow predefined or free-form data collection in a job, but not both

### ISSUE THREAD

**Issue by gino-m** [google/ground-platform#2075]
Summarizing @n-clinton: Having both leads to confusion and complexity in the survey creation process. If users need both, they can create two jobs ,one with predefined, one with free-form.

**Comment by kenstershiro** [comment]
Agreed to start from hero use cases to capture these more accurately - discussion is quite abstract atm

**Comment by gino-m** [comment]
It's not clear whether this or #1736 are needed. We need to nail down https://github.com/google/ground-platform/issues/2082 to be able to answer this.

**Comment by gino-m** [comment]
Counterpoint: multiple users have been confused by the "When adding a new site.." condition in the task list when setting up a new survey. If we require data collection jobs to be either structured (predefined)  or opportunistic (free form), we would remove this complexity, while still leaving open the option of using two jobs to allow both types of data collection. 

@vittorino @amegantz @rawbzz Any opinions here?

**Comment by gino-m** [comment]
> +1. It's not clear whether this or [#1736](https://github.com/google/ground-platform/issues/1736) are needed. We need to nail down [#2082](https://github.com/google/ground-platform/issues/2082) to be able to answer this.

I spoke with @lecrabe today, and he thinks #1736 is needed. I think #2082 is a larger effort, would prefer to focus on this FR for now.

**Comment by gino-m** [comment]
@rfontanarosa new design from @vittorino: 

![Image](https://github.com/user-attachments/assets/db89b338-1443-46c2-9015-592886a4c600)

**Comment by gino-m** [comment]
@vittorino Signal boost!
